### DISCUSSION TITLE
Why I got different results using automl.predict() and automl.model.predict()?

### DISCUSSION THREAD

**Discussion by azurechen97** [microsoft/FLAML#1282] [Q&A] [answered]
I trained a multi-classification LightGBM model via FLAML. But when I tested the model on the test set, I got completely different results on the same data by using `automl.predict()` and `automl.model.predict()`. What could be the problem?

**Comment by thinkall** [comment, marked as answer]
Hi @azurechen97, There may be some feature transformation in AutoML, in that case, `automl.predict()` will be different from `automl.model.predict()`. The code is [here](https://github.com/microsoft/FLAML/blob/main/flaml/automl/automl.py#L573-L574). You can pickle the trained automl instance and reuse it later. If you just want to use the `model.predict` (such as using mlflow autologging), you'll need to pickle the `feature_transformer` and `label_transformer`  and use it together with the saved model.

**Reply by azurechen97** [reply]
Thank you! The problem is solved perfectly.
